python/vision_explanation_methods/evaluation/pointing_game.py

```diff
@@ -148,15 +154,15 @@ def calculate_gt_salient_pixel_overlap(self,
         :return: return percent of salient pixel overlap with the ground truth
         :rtype: Float
         """
-        saliency_scores = torch.tensor(saliency_scores)
+        saliency_scores = torch.tensor(saliency_scores)[0]
         gt_bbox = torch.tensor(gt_bbox)
 
         gt_mask = torch.zeros_like(saliency_scores, dtype=torch.bool)
-        gt_mask[gt_bbox[0]:gt_bbox[2], gt_bbox[1]:gt_bbox[3]] = True
+        gt_mask[gt_bbox[1]:gt_bbox[3]+1, gt_bbox[0]:gt_bbox[2]+1] = True
 
         positive_mask = torch.gt(saliency_scores, 0)
         positive_gt_mask = torch.logical_and(positive_mask, gt_mask)
-        good = positive_gt_mask.sum()
-        total = positive_mask.sum()
+        good = torch.count_nonzero(positive_gt_mask)
+        total = torch.count_nonzero(positive_mask)
```

Comment:
Wouldnt the gt_mask and not the positive mask? The denominator is the issue when getting the values.

Reply:
Right now the return value is # of salient pixels in the bounding box / # of total salient pixels. 

postive_mask is essentially the same as gt_mask. 

Does that answer your question?

In the future, I can add additional metrics with different denomenators (ie divide by # of pixels in the gt bounding box, divide by the # of pixels in the segmented object, etc.)

Reply:
This would not be the case. The denominator how you provided is the saliency scores thresholded at 0. This will result in the pointing game having a higher accuracy compared to the denominator of the gt_mask.

Reply:
Right that makes sense, but in gt_mask there are artificially set "-1"s --> why would we want those to contribute to the sum?

Reply:
The bounding box will not always be the best, but with having the denominator be the saliency score it will be skewed higher and the saliency may not reflect the object. In the bear example that I tried this, it gives a higher accuracy then what it is and in the bounding box provided it covers a good amount of the bear.

Reply:
and with gt_mask created with torch.zeros_like, there should be no -1s?

Reply:
I think I understand what you are saying now. I was confusing gt_mask and saliency_scores. 

I updated the code to calculate the following: number of salient pixels in the gt bbox / number of pixels in the gt bbox

LMK if the code looks good to you.

Reply:
Sounds good.